src/msw/window.cpp

```diff
@@ -3766,40 +3777,43 @@ wxWindowMSW::MSWHandleMessage(WXLRESULT *result,
 
                     wxUxThemeHandle hTheme((const wxWindow *)this, L"EDIT");
                     wxWindowDC dc((wxWindow *)this);
-                    wxMSWDCImpl *impl = (wxMSWDCImpl*) dc.GetImpl();
+                    wxMSWDCImpl *impl = (wxMSWDCImpl*)dc.GetImpl();
 
                     // Clip the DC so that you only draw on the non-client area
-                    RECT rcBorder;
-                    wxCopyRectToRECT(GetSize(), rcBorder);
-
                     RECT rcClient;
-                    ::GetThemeBackgroundContentRect(
-                        hTheme, GetHdcOf(*impl), EP_EDITTEXT, ETS_NORMAL, &rcBorder, &rcClient);
-                    InflateRect(&rcClient, -1, -1);
+                    wxCopyRectToRECT(GetSize(), rcClient);
 
-                    ::ExcludeClipRect(GetHdcOf(*impl), rcClient.left, rcClient.top,
-                                      rcClient.right, rcClient.bottom);
+                    RECT rcContent = rcClient;
+                    //::CopyRect(&rcContent, &rcClient);
 
-                    // Make sure the background is in a proper state
-                    if (::IsThemeBackgroundPartiallyTransparent(hTheme, EP_EDITTEXT, ETS_NORMAL))
-                    {
-                        ::DrawThemeParentBackground(GetHwnd(), GetHdcOf(*impl), &rcBorder);
-                    }
-
-                    // Draw the border
+                    // Get the state ID
                     int nState;
-                    if ( !IsEnabled() )
+
+                    if (!IsEnabled())
                         nState = ETS_DISABLED;
-                    // should we check this?
-                    //else if ( ::GetWindowLong(GetHwnd(), GWL_STYLE) & ES_READONLY)
+                    // Should we check this?
+                    //else if (::GetWindowLong(GetHwnd(), GWL_STYLE) & ES_READONLY)
                     //    nState = ETS_READONLY;
                     else
                         nState = ETS_NORMAL;
-                    ::DrawThemeBackground(hTheme, GetHdcOf(*impl), EP_EDITTEXT, nState, &rcBorder, NULL);
+
+                    ::GetThemeBackgroundContentRect(hTheme, GetHdcOf(*impl), EP_EDITTEXT,
```

Comment:
Could you please explain what exactly has changed here? I see that the code has been moved around, but I don't understand why was it done and what has actually changed.

Reply:
In the original code:
`RECT rcClient;

::GetThemeBackgroundContentRect(hTheme, GetHdcOf(*impl), EP_EDITTEXT, ETS_NORMAL, &rcBorder, &rcClient);`

1st) `rcClient` was never initialized, and therefore if `GetThemeBackgroundContentRect` fails it will not alter `rcClient` at all, and `rcClient` will have a hollow value, thats why I initialized it in the fixed one: `RECT rcContent = rcClient;`
2nd) I moved the order so that: `GetThemeBackgroundContentRect` will be executed only after I get the state of the ID and use it for calculation. Because in the original code, `GetThemeBackgroundContentRect` only used `ETS_NORMAL` in any case.